### ISSUE TITLE
Fix the way MongoClient is created during parallelization

### ISSUE THREAD

**Issue by sirloon** [biothings/biothings.api#46]
See warning from pymongo: 

UserWarning: MongoClient opened before fork. Create MongoClient only after forking. See PyMongo's docume
ntation for details: http://api.mongodb.org/python/current/faq.html#is-pymongo-fork-safe                                                                                                             
  "MongoClient opened before fork. Create MongoClient only "

**Comment by zcqian** [comment]
Temporarily marking this as won't fix.

To prevent this warning, BioThings Hub must not use fork when creating child processes -- i.e. new processes must be spawned instead of forked. However, many data sources rely on forking to behave properly. This is because Dumper runs some tasks in a new process, and some of these tasks involve components that are dynamically generated (for instance, the actual code for manifest based plugins). In a forked process, these resources are directly available for use, but in a spawned process, it needs to be importable or serializable for inter-process communication, but dynamically generated code and some other resources are not importable or serializable.

Right now the biggest side effect of forking the threaded main process is that forked processes hang when having a debugger attached and hitting a breakpoint, aside from the warning from PyMongo. Changing the default multiprocessing context from fork to spawn will cause a lot of data sources to break, which is highly undesirable.

Also as a side note, if forking is needed on macOS hosts, set `OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` in the environment when running the Hub.
